### PR TITLE
Allow plugins to change decompiled file extensions

### PR DESCRIPTION
**Pull request by sschr15**

### CODE REVIEW

File: src/org/jetbrains/java/decompiler/api/plugin/LanguageSpec.java [hunk 1]
```diff
@@ -8,12 +8,18 @@ public final class LanguageSpec {
   public final GraphParser graphParser;
   public final StatementWriter writer;
   public final Pass pass;
-
+  public final String extension;
+  
   public LanguageSpec(String name, LanguageChooser chooser, GraphParser graphParser, StatementWriter writer, Pass pass) {
+    this(name, chooser, graphParser, writer, pass, "java");
```
Comment (jaskarth) [inline src/org/jetbrains/java/decompiler/api/plugin/LanguageSpec.java:14]: I think this should be removed. It's technically a breaking change but I don't think anyone else has written language plugins yet and defaulting to java is not ideal.

File: testFixtures/org/jetbrains/java/decompiler/SingleClassesTestBase.java [hunk 2]
```diff
@@ -247,14 +268,14 @@ public void run(Object[] options, SingleClassesTestBase base) throws IOException
           throw new UncheckedIOException(e);
         }
         //noinspection ConstantConditions
-        assumeTrue(false, referenceFile.getFileName() + " was not present yet");
+        abort(referenceFile.getFileName() + " was not present yet");
```
Comment (jaskarth) [inline testFixtures/org/jetbrains/java/decompiler/SingleClassesTestBase.java:271]: What's the reasoning behind this change?

File: testFixtures/org/jetbrains/java/decompiler/SingleClassesTestBase.java [hunk 3]
```diff
@@ -220,16 +228,29 @@ public void run(Object[] options, SingleClassesTestBase base) throws IOException
 
       String testFileName = classFile.getFileName().toString();
       String testName = testFileName.substring(0, testFileName.length() - 6);
-      Path decompiledFile = fixture.getTargetDir().resolve(testName + ".java");
-      assertTrue(Files.isRegularFile(decompiledFile));
+
+      Path decompiledFile;
+      if (expectedFileName == null) {
```
Comment (jaskarth) [inline testFixtures/org/jetbrains/java/decompiler/SingleClassesTestBase.java:233]: I think this logic is fairly complex now, it needs some comments.

**Review by jaskarth — CHANGES_REQUESTED** [review]
Thanks for doing this! I've added some comments here.

File: testFixtures/org/jetbrains/java/decompiler/SingleClassesTestBase.java (hunk 2 above)
Comment (sschr15) [reply testFixtures/org/jetbrains/java/decompiler/SingleClassesTestBase.java]: The reasoning was to follow the `fail` code I had above, but I've reverted that with my new commit

**Review by jaskarth — APPROVED** [review]
Looks good now, thanks!

**Pull request merged**